pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -238,228 +164,82 @@ func init() {
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
 func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
 
-// UUID gets the BIOS UUID via the sys interface.  This UUID is known by vsphere
-func getvmUUID() (string, error) {
-	id, err := ioutil.ReadFile(UUIDPath)
-	if err != nil {
-		return "", fmt.Errorf("error retrieving vm uuid: %s", err)
-	}
-	uuidFromFile := string(id[:])
-	//strip leading and trailing white space and new line char
-	uuid := strings.TrimSpace(uuidFromFile)
-	// check the uuid starts with "VMware-"
-	if !strings.HasPrefix(uuid, UUIDPrefix) {
-		return "", fmt.Errorf("Failed to match Prefix, UUID read from the file is %v", uuidFromFile)
-	}
-	// Strip the prefix and while spaces and -
-	uuid = strings.Replace(uuid[len(UUIDPrefix):(len(uuid))], " ", "", -1)
-	uuid = strings.Replace(uuid, "-", "", -1)
-	if len(uuid) != 32 {
-		return "", fmt.Errorf("Length check failed, UUID read from the file is %v", uuidFromFile)
-	}
-	// need to add dashes, e.g. "564d395e-d807-e18a-cb25-b79f65eb2b9f"
-	uuid = fmt.Sprintf("%s-%s-%s-%s-%s", uuid[0:8], uuid[8:12], uuid[12:16], uuid[16:20], uuid[20:32])
-	return uuid, nil
-}
-
-// Returns the name of the VM on which this code is running.
-// Will attempt to determine the machine's name via it's UUID in this precedence order, failing if neither have a UUID:
-// * cloud config value VMUUID
-// * sysfs entry
-func getVMName(client *govmomi.Client, cfg *VSphereConfig) (string, error) {
-	var vmUUID string
-	var err error
-
-	if cfg.Global.VMUUID != "" {
-		vmUUID = cfg.Global.VMUUID
-	} else {
-		// This needs root privileges on the host, and will fail otherwise.
-		vmUUID, err = getvmUUID()
-		if err != nil {
-			return "", err
-		}
-		cfg.Global.VMUUID = vmUUID
-	}
-
-	if vmUUID == "" {
-		return "", fmt.Errorf("unable to determine machine ID from cloud configuration or sysfs")
-	}
-
-	// Create context
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	// Create a new finder
-	f := find.NewFinder(client.Client, true)
-
-	// Fetch and set data center
-	dc, err := f.Datacenter(ctx, cfg.Global.Datacenter)
-	if err != nil {
-		return "", err
-	}
-	f.SetDatacenter(dc)
-
-	s := object.NewSearchIndex(client.Client)
-
-	svm, err := s.FindByUuid(ctx, dc, strings.ToLower(strings.TrimSpace(vmUUID)), true, nil)
-	if err != nil {
-		return "", err
-	}
-
-	if svm == nil {
-		return "", fmt.Errorf("unable to find machine reference by UUID")
-	}
-
-	var vm mo.VirtualMachine
-	err = s.Properties(ctx, svm.Reference(), []string{"name"}, &vm)
-	if err != nil {
-		return "", err
-	}
-
-	return vm.Name, nil
-}
-
 func newVSphere(cfg VSphereConfig) (*VSphere, error) {
```

Comment:
What this function is all about?
Please add header comment.

newVSphere is not intuitive name. Can you please comeup with some good name?